mlrun/runtimes/daskjob.py

```diff
@@ -457,7 +457,7 @@ def _run(self, runobj: RunObject, execution):
         handler = runobj.spec.handler
         self._force_handler(handler)
 
-        extra_env = self._generate_runtime_env(runobj)
+        extra_env = self.generate_runtime_env(runobj)
         environ.update(extra_env)
```

Comment:
not related, but this looks like a big hazard. is overrides the environ with runtime-specific envvars?
if code runs (and it might is) on server side, it is exposed to race condition of overridden envvars. im not sure where it is being used afterwards, but lets add a TODO here to make sure we understand it does not run on server-side.

Reply:
Agreed. I considered removing this code, because it seems to me that its original intention was to mirror the responsibility split employed on the creation of functions with the type Job. In such case, it falls to `_run` the role of acquiring the runtime variables (see [here](https://github.com/mlrun/mlrun/blob/development/mlrun/runtimes/kubejob.py#L317C44-L317C44) and [here](https://github.com/mlrun/mlrun/blob/development/mlrun/runtimes/base.py#L424)).

During my testing, however, while `_run` executes in the MLRun API Pod for functions of type Job, it does not on functions of the type Dask. In this case, `_run` seem to only run in the Dask Scheduler Pod, so it's already too late to get the traditional runtime variables (and also too late to incorrectly overwrite the environment on MLRun API Pods). That's why `MLRUN_DBPATH` wasn't available originally for Dask functions.

I added a comment with a TODO here so we can reconsider this code later. Should I create a Jira task as well?

Reply:
🚀 TIL
Yes please!

Reply:
For future reference, this discussion will be followed up on [ML-4515](https://jira.iguazeng.com/browse/ML-4515).